server/util/usageutil/usageutil.go

```diff
@@ -4,10 +4,56 @@ import (
 	"context"
 
 	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/bazel_request"
+	"github.com/buildbuddy-io/buildbuddy/server/util/pbwireutil"
+	"google.golang.org/grpc/metadata"
+)
+
+// Client label constants.
+const (
+	bazelClientLabel    = "bazel"
+	executorClientLabel = "executor"
 )
 
 // Labels returns usage labels for the given request context.
 func Labels(ctx context.Context) (*tables.UsageLabels, error) {
-	// TODO: populate with real values.
-	return &tables.UsageLabels{}, nil
+	return &tables.UsageLabels{
+		Origin: originLabel(ctx),
+		Client: clientLabel(ctx),
+	}, nil
+}
+
+func originLabel(ctx context.Context) string {
+	vals := metadata.ValueFromIncomingContext(ctx, "x-buildbuddy-origin")
+	if len(vals) == 0 {
+		return ""
+	}
+	return vals[0]
+}
+
+func clientLabel(ctx context.Context) string {
+	// Note: we avoid deserializing the RequestMetadata proto here since
+	// proto deserialization is too expensive to run on every request.
+	b := bazel_request.GetRequestMetadataBytes(ctx)
```

Comment:
Do we do this with metadata in multiple places? Would it make sense to deserialize the full proto once in the interceptor and throw fields in the context so we don't need to reparse?

Reply:
Yeah, that would be a nice improvement - will look at that when I have more time.